pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -177,8 +177,8 @@ func TestPriorityQueue_AddWithReversePriorityLessFunc(t *testing.T) {
 func TestPriorityQueue_AddUnschedulableIfNotPresent(t *testing.T) {
 	q := createAndRunPriorityQueue(newDefaultQueueSort())
 	q.Add(&highPriNominatedPod)
```

Comment:
can we pop instead?

Reply:
Pop() is not always applicable 🤷‍♂️

Reply:
I think what we have now is not much different from what we had with the extra flag.

In reality, we don't need to test calling `AddUnschedulableIfNotPresent` in conditions we don't need. The only usage of the function (other than tests) is from `MakeDefaultErrorFunc`. So we should only be testing `AddUnschedulableIfNotPresent` with the condition that `Pop` was called first.

That said, a second call to `AddUnschedulableIfNotPresent` should just work without the wrapper you added. Doesn't it?

Reply:
> That said, a second call to AddUnschedulableIfNotPresent should just work without the wrapper you added. Doesn't it?

Correct, as I mentioned at L165~166:

```
// NOTE: if the unit test uses Pop() to pop a Pod, you can directly call AddUnschedulableIfNotPresent()
// as Pop() already put the popped Pod to q.processingPods.
```

Reply:
> In reality, we don't need to test calling AddUnschedulableIfNotPresent in conditions we don't need.

True, but the existing tests are not composed this way. And cleaning them up may not be trivial. So I just leave them as is for now.